src/Glpi/Controller/AbstractController.php

```diff
@@ -0,0 +1,75 @@
+<?php
+
+/**
+ * ---------------------------------------------------------------------
+ *
+ * GLPI - Gestionnaire Libre de Parc Informatique
+ *
+ * http://glpi-project.org
+ *
+ * @copyright 2015-2024 Teclib' and contributors.
+ * @licence   https://www.gnu.org/licenses/gpl-3.0.html
+ *
+ * ---------------------------------------------------------------------
+ *
+ * LICENSE
+ *
+ * This file is part of GLPI.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ *
+ * ---------------------------------------------------------------------
+ */
+
+namespace Glpi\Controller;
+
+use Glpi\Application\View\TemplateRenderer;
+use Glpi\DependencyInjection\PublicService;
+use Symfony\Component\HttpFoundation\Response;
+
+abstract class AbstractController implements PublicService
+{
+    /**
+     * Helper method to get a response containing the content of a rendered
+     * twig template.
+     *
+     * @param string $view Path to a twig template, which will be looked for in
+     * the "templates" folder.
+     * For example, "my_template.html.twig" will be resolved to `templates/my_template.html.twig`.
+     * @param array $parameters The expected parameters of the twig template.
+     * @param Response $response Optional parameter which serves as the "base"
+     * response into which the renderer twig content will be inserted.
+     * You should only use it if you need to set some specific headers into the
+     * response or to set an http return code different than 200.
+     *
+     * @return Response
+     */
+    final protected function render(
+        string $view,
+        array $parameters = [],
+        Response $response = new Response(),
+    ): Response {
+        $twig = TemplateRenderer::getInstance();
+
+        // We must use output buffering here as Html::header() and Html::footer()
+        // output content directly.
+        // TODO: fix header() and footer() methods and remove output buffering
+        ob_start();
+        $twig->display($view, $parameters);
+        $content = ob_get_clean();
```

Comment:
Isn't Twig's `render()` method be more appropriate?

```suggestion
        $content = $twig->render($view, $parameters);
```

Reply:
No, it does not work because the `skeleton.html.twig` file is calling `Html::header()` and `Html::footer()` which echo directly some content.

Once we fix or remove these methods we will indeed be able to remove this ugly workaround (thus the TODO block above).

Reply:
It's clear, thanks for the details 👍